### PR TITLE
Fix spack install --v[tab] spec

### PR DESCRIPTION
**Pull request by haampie**
Currently `spack install --v[tab] spec` completes to `spack install - spec`.

This regression was caused by #14393

### CODE REVIEW

**Review by tgamblin — DISMISSED** [review]

**Comment by haampie** [comment]
@tgamblin can't find your review anymore (github glitch?) but I think there was just an off by one error that's now fixed

**Review by tgamblin — APPROVED** [review]

**Comment by haampie** [comment]
thanks for merging, this bugged me for a long time

**Comment by sethrj** [comment]
Github has a habit of erasing old comments when you force-push.

**Comment by tgamblin** [comment]
Any new commits invalidate approvals -- for security, because we allow auto-merge.  Without that, you could push random stuff after approval and have it merged.

**Comment by sethrj** [comment]
I thought @haampie was referring to the missing comments? If you push a new commit without forcing, it'll invalidate the approval but at least allow you to look for the comments made as part of reviews of previous commits. If you force-push, those comments seem to become inaccessible. (I often find this out when clicking notifications: github says it "looked everywhere" but can't find a commit.)

**Comment by haampie** [comment]
Yeah, I thought Todd left a comment, but maybe that was not the case :)

**Pull request merged**
